core/tee/tee_rpmb_fs.c

```diff
@@ -437,12 +437,14 @@ static void tee_rpmb_free(struct tee_rpmb_mem *mem)
 	if (!mem)
 		return;
 
-	thread_rpc_free_payload(mem->phreq_cookie);
-	thread_rpc_free_payload(mem->phresp_cookie);
-	mem->phreq = 0;
-	mem->phreq_cookie = 0;
-	mem->phresp = 0;
-	mem->phresp_cookie = 0;
+	if (mem->phreq) {
```

Comment:
Yeah... We're not likely to ever see 0 as a valid physical address inside OP-TEE I suppose, but you are of course right. Note that to be able to handle address zero, we would need additional compile flags (by default, GCC assumes it is never valid to de-reference a null pointer).

Anyway, I see two options that would be more correct here:
- Assume nothing about the values of the physical address or the cookie, and add a return code to `thread_rpc_alloc_payload()`. Then add flags to `struct tee_rpmb_mem` to keep track of which alloc is valid. A bit overkill maybe?
- Update the interface documentation to state that the success condition is (*cookie != 0) on return, not (*payload != 0), and update the test accordingly.

Which one do you prefer?